functions.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * connects to the database and extracts all items required and returns in an associative array
+ * @param $dbName, the name of the db to connect to
+ *
+ * @return array, the array of items in the collection
+ */
+function connect_to_db_and_extract_items($dbName)
+{
+    $db = new PDO('mysql:host=db;dbname=' . $dbName, 'root', 'password'); //initialise the db connection
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    $query = $db->prepare('SELECT `variety`, `tones`, `cost`, `nameOfBrand`, `country`, `wine`.`img_location` FROM `wine` JOIN `brands` ON `wine`.`brand_id` = `brands`.`id` JOIN `country` ON `wine`.`country_of_origin`=`country`.`id`;');
+    $query->execute();
+    return $query->fetchAll();
+}
+
+/**
+ * takes items from an array and returns the values within HTML
+ * @param array $dataFromQuery, array of data from database
+ *
+ * @return string, return string of html
+ */
+
+function addItemToHTML(array $dataFromQuery): string
+{
+    $result = '';
+    foreach ($dataFromQuery as $itemFromQuery) {
+        if ((!isset($itemFromQuery['variety'])) || (!isset($itemFromQuery['nameOfBrand'])) || (!isset($itemFromQuery['cost'])) || (!isset($itemFromQuery['country'])) || (!isset($itemFromQuery['tones']))) {
+            $result .= 'Error can\'t find key';
```

Comment:
if you pass in an array where the first sub array is valid, the second one is not and the third one is valid, you will get a very odd result from this... try it

Reply:
it prints the error to the screen inbetween the valid sub array, so I think I should break the loop if there is an error?

Reply:
i changed my mind, and have outputted from the function an array that contains the result of html and the error so they're not concatenated